module/sell.py

```diff
@@ -13,47 +13,50 @@ def sell(update: Update, context: CallbackContext) -> None:
                     context.bot.send_message(chat_id, "Il libro è: \nTitolo: \"" + title_and_author + "\"")
-                    title  = soup.find('strong').text.split('/')[0]
-                    authors  = soup.find('strong').text.split('/')[1]
-                    isbn = str(soup.findAll('td')).split("bibInfoData")[len(str(soup.findAll('td')).split("bibInfoData"))-1].split("\n")[1].split("<")[0]
                     df = get_on_sale()
                     new_row = pd.DataFrame({'ISBN':str(isbn),'Titolo':str(title), 'Autori':str(authors), 'Venditore':str(username), 'Prezzo':str(prezzo)}, index=[len(df)])
                     df = pd.concat([df, new_row])
                     df.to_csv(SHOP_DB_PATH)
-                    new_book = pd.DataFrame({'ISBN':str(isbn),'Titolo':str(title), 'Autori':str(authors)}, index=[len(books)])
-                    books = pd.concat([books, new_book])
-                    books.to_csv(BOOKS_DB_PATH)
                     context.bot.send_message(chat_id, "Il libro è stato aggiunto al database.")
                 else:
-                    context.bot.send_message(chat_id, "Libro non trovato. Controlla di aver inserito correttamente l'ISBN")
-        except Exception as e:
-            print(str(e))
-            context.bot.send_message(chat_id, "Prezzo non valido")
+                    url="https://catalogo.unict.it/search/i?SEARCH=" + user_isbn + "&sortdropdown=-&searchscope=9"
+                    x = requests.get(url)
+                    soup = BeautifulSoup(x.content, 'html.parser')
+                    check = str(soup.findAll('td'))
+                    if "No matches found" not in check:
+                        title_and_author  = soup.find('strong').text
+                        context.bot.send_message(chat_id, "Il libro è: \nTitolo: \"" + title_and_author + "\"")
+                        title  = soup.find('strong').text.split('/')[0]
+                        authors  = soup.find('strong').text.split('/')[1]
+                        isbn = str(soup.findAll('td')).split("bibInfoData")[len(str(soup.findAll('td')).split("bibInfoData"))-1].split("\n")[1].split("<")[0]
+                        df = get_on_sale()
+                        new_row = pd.DataFrame({'ISBN':str(isbn),'Titolo':str(title), 'Autori':str(authors), 'Venditore':str(username), 'Prezzo':str(prezzo)}, index=[len(df)])
+                        df = pd.concat([df, new_row])
+                        df.to_csv(SHOP_DB_PATH)
+                        new_book = pd.DataFrame({'ISBN':str(isbn),'Titolo':str(title), 'Autori':str(authors)}, index=[len(books)])
+                        books = pd.concat([books, new_book])
+                        books.to_csv(BOOKS_DB_PATH)
+                        context.bot.send_message(chat_id, "Il libro è stato aggiunto al database.")
+                    else:
+                        context.bot.send_message(chat_id, "Libro non trovato. Controlla di aver inserito correttamente l'ISBN.")
```

Comment:
what about putting all this of the below `else` in a separate function like `parsing_books_from_unict()`?
The function could return a boolean value and a possible result could be
```python
                    if parsing_books_from_unict():
                        context.bot.send_message(chat_id, "Il libro è stato aggiunto al database.")
                    else:
                        context.bot.send_message(chat_id, "Libro non trovato. Controlla di aver inserito correttamente l'ISBN.")
```

Reply:
Refactor done!